torch/_dynamo/skipfiles.py

```diff
@@ -29,31 +29,114 @@
 import typing
 import unittest
 import weakref
+from typing import Optional
 
 import torch
 import torch._inductor.test_operators
 import torch.distributed
 import torch.utils._content_store
 
-from . import comptime, config, external_utils
+from . import comptime, external_utils
 
 """
 A note on skipfiles:
 
-Dynamo consults this file to determine whether code should be compiled or skipped.
+Dynamo consults this file to determine whether function should be inlined or skipped.
 
 A skip applies at the frame boundary, meaning dynamo either triggers a graph break
-at the beginning of the frame or attempts to trace the whole frame.  When skipping
+at the beginning of the frame or attempts to trace/inline the whole frame. When skipping
 a frame, recursively called frames are still traced by dynamo unless also skipped.
 
 Skipfiles (skipped at the file level instead of function level) still apply on a
 frame-by-frame boundary as dynamo traces, but apply to all functions in that file.
 
 @skip is a helper decorator that can be applied to your function to cause it to be
 included here.
+
+Dynamo skip/inline rules & priorities are defined as follows:
+* Inline is the default behavior and will be used unless explicitly skipped.
+* Dynamo has two SKIPLIST: BUILTIN_SKIPLIST and THIRDPARTY_SKIPLIST.
+    * BUILTIN_SKIPLIST contains builtin python modules, such as abc, collections, etc.
+    * THIRDPARTY_SKIPLIST contains common third party libraries, such as numpy, pandas, etc.
+* Functions in these two SKIPLISTs are always skipped, except when they are explicitly
+    put into the two INLINELIST: FILENAME_INLINELIST and SUBMODULE_INLINELIST.
+* PyTorch(torch) is in the BUILTIN_SKIPLIST by default, but there are many cases
+    where we want inline the functions under torch namespace. We should add them
+    into FILENAME_INLINELIST or SUBMODULE_INLINELIST to make dynamo inline those functions.
```

Comment:
Also, torch will typically not actually induce a graph break when you're dynamo'ing because there's built-in behavior that overrides the skip behavior (TorchVariable)